src/integration-tests/breakpointModes.spec.ts

```diff
@@ -0,0 +1,279 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *********************************************************************/
+import * as path from 'path';
+import * as os from 'os';
+import { expect } from 'chai';
+import { CdtDebugClient } from './debugClient';
+import {
+    standardBeforeEach,
+    testProgramsDir,
+    fillDefaults,
+    isRemoteTest,
+    gdbAsync,
+} from './utils';
+import { DebugProtocol } from '@vscode/debugprotocol';
+import { MIBreakpointMode } from '../mi';
+
+// This mock adapter is overriding the getBreakpointOptions method.
+const adapter =
+    'integration-tests/mocks/debugAdapters/dynamicBreakpointOptions.js';
+const argHardwareBreakpointTrue = '--hardware-breakpoint-true';
+const argHardwareBreakpointFalse = '--hardware-breakpoint-false';
+const argBreakpointModeHardware = ['--breakpoint-mode', 'hardware'];
+const argBreakpointModeSoftware = ['--breakpoint-mode', 'software'];
+
+const program = 'count';
+
+const startDebugClientWithArgs = async (
+    test: Mocha.Test | undefined,
+    ...args: string[]
+) => {
+    const dc = await standardBeforeEach(adapter, args);
+    await dc.launchRequest(
+        fillDefaults(test, {
+            program: path.join(testProgramsDir, 'count'),
+        })
+    );
+    return dc;
+};
+
+const sendBreakpointRequest = (
+    dc: CdtDebugClient,
+    options?: Partial<DebugProtocol.SourceBreakpoint>
+) => {
+    return dc.setBreakpointsRequest({
+        source: {
+            name: `${program}.c`,
+            path: path.join(testProgramsDir, `${program}.c`),
+        },
+        breakpoints: [
+            {
+                column: 1,
+                line: 4,
+                ...(options ?? {}),
+            },
+        ],
+    });
+};
+
+const expectBreakpoint = async (
+    dc: CdtDebugClient,
+    response: DebugProtocol.SetBreakpointsResponse,
+    breakpointMode: MIBreakpointMode
+) => {
+    expect(response.body.breakpoints.length).eq(1);
+    expect(response.body.breakpoints[0].verified).eq(true);
+    expect(response.body.breakpoints[0].message).eq(undefined);
+    await dc.configurationDoneRequest();
+    // 'breakpoint-modified' output doesn't received in hardware breakpoints in Windows
+    if (os.platform() === 'win32' && breakpointMode === 'hardware') {
+        return;
+    }
+
+    let isCorrect;
+    let outputs;
+    while (!isCorrect) {
+        // Cover the case of getting event in Linux environment.
+        // If cannot get correct event, program timeout and test case failed.
+        outputs = await dc.waitForEvent('output');
+        isCorrect = outputs.body.output.includes('breakpoint-modified');
+    }
+    expect(outputs?.body.output).includes(
+        `type="${
+            breakpointMode === 'hardware' ? 'hw breakpoint' : 'breakpoint'
+        }"`
+    );
+};
+
+// Only do the hardware breakpoint tests in Windows in remote debugging with gdbAsync on
+const skipHardwareBreakpointTest =
+    os.platform() === 'win32' && (!isRemoteTest || !gdbAsync);
```

Comment:
Lets make this test simply skip on win32 until #351 is addressed. See https://github.com/eclipse-cdt-cloud/cdt-gdb-adapter/pull/350#issuecomment-2640501410 also

```suggestion
    os.platform() === 'win32';
```

Reply:
Updated.